### PR TITLE
code_806CD90

### PR DESCRIPTION
**Pull request by DizzyEggg**
WIP

### CODE REVIEW

**Comment by DizzyEggg** [comment]
Ready for review

**Review by SethBarberee — APPROVED** [review]

File: src/code_806CD90.c [hunk 1]
```diff
@@ -228,3 +255,761 @@ u32 sub_806CF98(Entity *entity)
...
+    }
+    else {
+        const u8 *str;
+
+        gFormatData_202DE30 = dmgStruct->dmg;
+        str = gUnknown_80F9764[dmgStruct->residualDmgType];
+
+        // Needed to match - the line can be safely removed
+        dmgStruct++;dmgStruct--;
+        //
```
Comment (AnonymousRandomPerson) [inline src/code_806CD90.c:631]: Empty comment.

File: src/code_806CD90.c [hunk 2]
```diff
@@ -228,3 +255,761 @@ u32 sub_806CF98(Entity *entity)
...
+        dmgStruct->unkF = 1;
+        return FALSE;
+    }
+
+    if (targetData->unk152 == 0) {
+        if (dmgStruct->isCrit) {
+            sub_80522F4(attacker, target, gUnknown_80F9614);
+        }
+        switch (dmgStruct->typeEffectiveness) {
+            case 0:
```
Comment (AnonymousRandomPerson) [inline src/code_806CD90.c:581]: There are `#define`s for these effectiveness constants in `type_chart.h`.

File: src/code_806CD90.c [hunk 3]
```diff
@@ -228,3 +255,761 @@ u32 sub_806CF98(Entity *entity)
...
+                }
+            }
+            else {
+                unkBool = FALSE;
+            }
+            if (targetData->eyesightStatus.eyesightStatus == STATUS_BLINKER || targetData->eyesightStatus.eyesightStatus == STATUS_CROSS_EYED)
+                unkBool = FALSE;
+            if (unkBool) {
+                EntityInfo *info = GetEntInfo(target);
+                info->action.direction = (GetDirectionTowardsPosition(&target->pos, &attacker->pos) & 7);
```
Comment (AnonymousRandomPerson) [inline src/code_806CD90.c:670]: The 7 can use `DIRECTION_MASK` from `constants/direction.h`.

**Comment by DizzyEggg** [comment]
Changes done

**Review by AnonymousRandomPerson — APPROVED** [review]

**Pull request merged**